Mathlib/CategoryTheory/Limits/Shapes/Pullbacks.lean

```diff
@@ -2746,6 +2752,29 @@ def baseChange [HasPullbacks C] {X Y : C} (f : X ⟶ Y) : Over Y ⥤ Over X wher
     apply Over.OverMorphism.ext; apply pullback.hom_ext
     · dsimp; simp
     · dsimp; simp
-#align category_theory.limits.base_change CategoryTheory.Limits.baseChange
-
-end CategoryTheory.Limits
+#align category_theory.limits.base_change CategoryTheory.Over.baseChange
+
+-- deprecated on 2024-05-15
+@[deprecated] noncomputable alias Limits.baseChange := Over.baseChange
+
+/-- The adjunction `Over.map ⊣ baseChange` -/
+@[simps! homEquiv_apply homEquiv_symm_apply unit_app counit_app]
+def mapAdjunction [HasPullbacks C] {X Y : C} (f : X ⟶ Y) : Over.map f ⊣ baseChange f :=
+  Adjunction.mkOfHomEquiv
+    { homEquiv := fun x y ↦
+        { toFun := fun u ↦ Over.homMk (pullback.lift u.left x.hom (by simp))
+          invFun := fun v ↦ Over.homMk (v.left ≫ pullback.fst) (by
+            dsimp
+            rw [Category.assoc, pullback.condition, ← Over.w v, Category.assoc]
+            dsimp)
+          left_inv := by aesop_cat
+          right_inv := fun v ↦ by
+            dsimp
+            ext
+            dsimp
+            ext
+            · simp
+            · rw [pullback.lift_snd, ← Over.w v]
+              dsimp }}
+
+end Over
```

Comment:
This is not required by the syntax, but it is slightly better to "close" the namespace.
```suggestion
end Over

end CategoryTheory
```